### PR TITLE
Creating example tests

### PR DESCRIPTION
**Pull request by KilimAnnejaro**
Per https://github.com/bazelbuild/bazel/pull/14432#issuecomment-994278938, adding example testing code to the sample repo.

@aiuto not sure if this is what you were looking for?

### CODE REVIEW

**Comment by KilimAnnejaro** [comment]
@aiuto @brandjon @c-parsons bump?

File: java-tutorial/BUILD [hunk 1]
```diff
@@ -1,6 +1,35 @@
...
 
 java_binary(
     name = "ProjectRunner",
+    srcs = ["src/main/java/com/example/ProjectRunner.java"],
+    main_class = "com.example.ProjectRunner",
+    deps = [":greeter",":calculator"],
+)
+
+java_library(
+    name = "java-maven-lib",
```
Comment (aiuto) [inline java-tutorial/BUILD:11]: Why java-*maven*-lib?  Isn't it just calculator_lib?

File: java-tutorial/BUILD [hunk 2]
```diff
@@ -1,6 +1,35 @@
...
     srcs = glob(["src/main/java/com/example/*.java"]),
```
Comment (aiuto) [inline java-tutorial/BUILD:12]: Doesn't this have to exclude ProjectRunner.java?

**Review by aiuto — COMMENTED** [review]
I'm going to pass this to someone who knows more about maven than I do. I never go near it if I can avoid it.

**Comment by aiuto** [comment]
Do we have a standard way to include junit without having to bring in maven
as well?
That would make a better example because it does not conflate java_test
with maven.

File: java-tutorial/BUILD [hunk 3]
```diff
@@ -1,6 +1,35 @@
...
+    test_class = "com.example.CalculatorTest",
+    deps = [
+        ":java-maven-lib",
+        "@maven//:junit_junit",
+    ],
+)
+
+java_library(
+    name = "greeter",
+    srcs = ["src/main/java/com/example/Greeting.java"],
```
Comment (jin) [inline java-tutorial/BUILD:27]: This doesn't exist..?

**Review by jin — CHANGES_REQUESTED** [review]
Thanks for your contribution!

This repo is used with https://docs.bazel.build/versions/main/tutorial/java.html. The BUILD file doesn't have the greeter target for that reason, because the instructions wants the user to add it manually. So I don't think we should change the contents of this repo at all, unless there's an intention to update the java tutorial on docs.bazel.build in tandem.

Also, there are already Junit + rules_jvm_external test examples in https://github.com/bazelbuild/examples/tree/main/java-maven (see https://github.com/bazelbuild/examples/blob/d1a1f1fcf460a7f3aa0eb845b22dab061eff8ec9/java-maven/BUILD#L17), so I'm not sure what this PR is adding.

**Comment by aiuto** [comment]
Going back to Jin's comment. This is part of the tutorial, so we don't want to change it without changing the tutorial in concert. I'm going to close this because it is so stale and someone can start again if they want to overhaul the whole tutorial.

**Pull request closed without merging**